_posts/2023-07-31-turbocharging-rust-code-verification.md

```diff
@@ -0,0 +1,405 @@
+---
+layout: post
+title:  "Turbocharging Rust Code Verification"
+---
+
+Kani is a verification tool that can help you systematically test properties about your Rust code.
+To learn more about Kani, check out the [Kani tutorial](https://model-checking.github.io/kani/kani-tutorial.html) and our [previous blog posts](https://model-checking.github.io/kani-verifier-blog/).
+
+Over the past 9 months we have optimised Kani at different levels to either improve general performance or solve some specific performance problems encountered on some user proof harnesses.
+
+In this post, we'll discuss three optimizations enabled through three different components of Kani. In short, by allowing the selection of the specific SAT solver to use for specify proof harnesses we obtain speedups of anywhere from 2x to 200x on specific proof harneses, and an 85% reduction in total runtime on a crate like `s2n-quic-core`; By adding a new goto program serializer that exports files in CBMC's own binary format we obtain a x4 speedup on the code generation + goto program export step; By improving constant propagation in CBMC for union types, Kani can now solve in a few seconds proof harnesses that used to explode in time and memory. Compared to the late 2022 version, the current version of Kani (August 2023) is swifter and solves more problems in less time.
```

Comment:
This is screaming "convert me into a bullet list" to me.
```suggestion
Over the past 9 months we have optimised Kani at different levels to either improve general performance or solve some specific performance problems encountered on some user verification harnesses.
In this post, we'll discuss three optimizations enabled through three different components of Kani. In short, by:
 - allowing the selection of the specific SAT solver to use for specify verification harnesses we obtain speedups of anywhere from 2x to 200x on specific verification harnesses, and an 85% reduction in total runtime on a crate like [s2n-quic-core](https://github.com/aws/s2n-quic/tree/main/quic/s2n-quic-core).
 - adding a new goto program serializer that exports files in CBMC's own binary format we obtain a x4 speedup on the code generation and goto program export steps.
 - improving constant propagation in CBMC for union types, Kani can now solve in a few seconds verification harnesses that used to explode in time and memory. Compared to the last version of 2022, the current version of Kani (August 2023) is swifter and solves more problems in less time.
```

Reply:
Note that there are other minor changes included:
 - replaced s2n-quic-core to be a link instead of code-formatted (`s2n-quic-core`)
 - rewording on last version of Kani
 - replace "proof" with "verification"

Reply:
replaced proof harness with verification harness globally